mlflow/utils/thread_utils.py

```diff
@@ -45,4 +46,10 @@ def set(self, value):
         Set a value for the thread-local variable.
         """
         self.thread_local.value = (value, os.getpid())
-        self._value_dict[threading.currentThread().ident] = value
+        self.__global_thread_values[threading.currentThread().ident] = value
```

Comment:
```suggestion
        self._global_thread_values[threading.currentThread().ident] = value
```

you don't need two underscores

Reply:
oh 2 underscores is suggested by @B-Step62 (for true private) , I am ok for either 1 or 2

Reply:
I'm actually inclined to two underscores to protect this more strongly - writing to this directly or mutating it can be a source of confusing bug. WDYT?

Reply:
makes sense, let's keep two underscore here.